DBSCE/Overview.md

```diff
@@ -0,0 +1,275 @@
+
+The device registration is expected to be a once-in-a-lifetime protected operation, and the user is expected to perform this operation with a clean room environment.
+
+![DeviceRegistration](./DeviceRegistration.svg)
+
+### Local Key Helper
+
+DBSC(E) introduces the concept of `Local Key Helper`.
+
+**Local Key Helper** is an integral part of the the **Device Registration Client**, a software interface responsible for the DBSC Key management. *Local key helper* can be public or private and is expected to be either shipped as a part of a given enterprise framework (with the IDP/OS) or can be installed by a provider in compliance with the protocol expanded below. DBSC(E) defines browser interaction with _Local key helpers_ for each platform. Those details are outlined in [KeyGeneration.md](./KeyGeneration.md).
+
+From the deployment point of view there are two types of local key helpers: _private_ and _public_
+
+- _Public local key helper_: Expected to have a well-documented API and can be used by any Identity Provider (IdP). Typically owned by a provider different from the IdP, communicates with the IdP as defined in DBSC(E) protocol. 
+- _Private local key helper_ : Is specific to an IdP. Can be only used by a specific IDP that owns the implementation and will have a private protocol to communicate with the IdP. 
+
+The Local Key Helper is responsible for:
+
+- Generation of the [binding key](#binding-key) and producing [binding statements](#binding-statement) (see below).
+- Producing signatures with the binding key.
+- Cleanup of the [binding key](#binding-key) and its artifacts (when the user clears the browser session or the key is unused for a long time).
+
+#### Platform Requirements
+
+This section prescribes the browser discovery process of a given local key helper for a few well known platforms:
+
+- [Windows](./LocalKeyHelper-Windows.md)
+- [MacOS](./LocalKeyHelper-Mac.md)
+- [Android](./LocalKeyHelper-Android.md)
+
+Note: Above are platform specifications for Local Key Helpers that can be used for DBSC(E) key generation. Any vendor can ship their local key helper in compliance with the DBSC(E).
+
+### Attestation Service
+
+Attestation service is responsible for providing the attestation statement for the binding key. The attestation service can be owned by the IdP or a third party. DBSC(E) relies on the attestation service to enable the IDP to validate the binding statement and ensure that the binding key and the device key belong to the same device. An attestation service can be part of the IdP, or a separate service. 
+
+This document does not define the implementation details of the Attestation Service. It defines the artifacts which are generated during the [Device Registration](#device-registration-client) and are necessary for the Attestation Service to validate the binding statement.
```

Comment:
>>> and are necessary for the Attestation Service to validate the binding statement.

Should we skip "attestation service" here, since IDP will validate binding statement?

Reply:
Done